perfkitbenchmarker/pkb.py

```diff
@@ -280,10 +280,13 @@ def RunBenchmark(benchmark, collector, sequence_number, total_benchmarks,
           # a reference to the spec in order to delete it in the "finally"
           # section below.
           spec = benchmark_spec.BenchmarkSpec(benchmark_config, benchmark_uid)
+          vm_util.SetThreadBenchmarkSpec(spec)
+          spec.DoFlagsMerge()
           spec.ConstructVirtualMachines()
           DoPreparePhase(benchmark, benchmark_name, spec, detailed_timer)
         else:
           spec = benchmark_spec.BenchmarkSpec.GetSpecFromFile(benchmark_uid)
+          vm_util.SetThreadBenchmarkSpec(spec)
```

Comment:
Would it be possible to set the current thread benchmark spec during construction?